docs/lang/articles/contribution/development_tips.md

```diff
@@ -0,0 +1,102 @@
+---
+sidebar_position: 3
+---
+
+# Development tips
+
+This section gives you some tips on the Taichi compiler development.
```

Comment:
nit: tips on Taichi development? In the future we might just dump all dev tips here :D

Reply:
Well as this is in the `Contribution Guide` folder, I think it needs to focus on developing the Taichi compiler. Tips regarding using Taichi should be in user guides in other folders.